app/grandchallenge/evaluation/admin.py

```diff
@@ -78,13 +84,24 @@ class PhaseAdmin(admin.ModelAdmin):
         "algorithm_outputs",
         "archive",
     )
-    readonly_fields = ("give_algorithm_editors_job_view_permissions",)
+    readonly_fields = (
+        "give_algorithm_editors_job_view_permissions",
+        "challenge_forge_json",
+    )
     form = PhaseAdminForm
 
     @admin.display(boolean=True)
     def open_for_submissions(self, instance):
         return instance.open_for_submissions
 
+    @staticmethod
+    def challenge_forge_json(obj):
+        json_desc = get_forge_json_description(
+            challenge=obj.challenge,
+            phases_queryset=Phase.objects.filter(id=obj.id),
```

Comment:
Small improvement is to avoid a second lookup here. You could instead do:

```python
json_desc = get_forge_json_description(
             challenge=obj.challenge,
             phase_pks=[obj.pk], # FYI it is best to avoid using id as that is overloaded, pk is more explicit that you're dealing with a Django object
)

...

def get_forge_json_description(challenge, phase_pks=None):
     """
     Generates a JSON description of the challenge and phases suitable for
     grand-challenge-forge to generate a challenge pack.
     """
     
     # FYI going through the reverse relation means that we know that all the phases belong to the challenge
     phases = challenge.phase_set.filter(
         archive__isnull=False,
         submission_kind=SubmissionKindChoices.ALGORITHM,
     ).prefetch_related("archive", "inputs", "outputs")

     if phase_pks is not None:
         phases = phases.filter(pk__in=phase_pks)

     ...
```

Reply:
The `pk` versus `id` is a good point here. In a silly habit, I looked at the model and straight up read the `id`. 

About the second lookup, to clarify, I thought query sets were lazily evaluated and passing an unevaluated one could be filtered before it would be.

Reply:
Yes, but this way we ensure that the phases passed actually belong to the challenge so it is internally consistent.